CSE/CSE/GraphFeature/Graph.cs

```diff
@@ -0,0 +1,155 @@
+            var titles1 = new System.Windows.Forms.DataVisualization.Charting.Title
+            {
+                Name = titles,
+                Text = titles + " Graph Data",
+                Visible = true,
+            };
+
+            this.chart1.Titles.Add(titles1);
+        }
+
+        private void ChartLegends(string name)
+        {
+            var legend1 = new System.Windows.Forms.DataVisualization.Charting.Legend
+            {
+                Name = name,
+            };
+            this.chart1.Legends.Add(legend1);
+        }
+
+        private void ChartSeries(string name)
+        {
+            var series1 = new System.Windows.Forms.DataVisualization.Charting.Series
+            {
+                XValueType = ChartValueType.DateTime,
+                Name = name,
+                Color = System.Drawing.Color.Red,
+                BorderWidth = 5,
+                IsVisibleInLegend = true,
+                ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Line,
+            };
+
+            foreach (var values in dateAndPrice)
+            {
+                series1.Points.AddXY(values.Key, values.Value);
+            }
+
+
+            this.chart1.Series.Add(series1);
+            this.ChartLegends(name);
+        }
+
+        private void ChartSeriesUpdate(string name)
+        {
+            var series1 = new System.Windows.Forms.DataVisualization.Charting.Series
+            {
+                XValueType = ChartValueType.DateTime,
+                Name = name,
+                Color = System.Drawing.Color.Red,
+                BorderWidth = 5,
+                IsVisibleInLegend = true,
+                ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Line,
+            };
+        }
+
+        private void Graph_Load(object sender, EventArgs e)
+        {
+            System.Windows.Forms.Cursor.Current = Cursors.WaitCursor;
+            this.RunReport();
+            System.Windows.Forms.Cursor.Current = Cursors.Default;
+        }
+
+        private void RunReport()
+        {
+            //Clear Chart
+            this.chart1.Series.Clear();
+            this.chart1.Legends.Clear();
+            this.chart1.ChartAreas.Clear();
+            this.chart1.Titles.Clear();
+
+            //Build Chart
+            this.ChartSeries(productName);
+            this.ChartAreas("Price");
+            this.ChartTitle(productName);
+            this.chart1.Invalidate();
+        }
+
+        private void graphComboBox_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            switch (graphComboBox.Text)
+            {
+                case "Years":
+                    this.chart1.ChartAreas[0].AxisX.IntervalType = DateTimeIntervalType.Years;
+                    break;
+                case "Months":
+                    this.chart1.ChartAreas[0].AxisX.IntervalType = DateTimeIntervalType.Months;
+                    break;
+                case "Days":
```

Comment:
You can ommit case "Days" as you fave exactly the same case in default.

Reply:
This relates with changing graph's scale and it  will be done in the near future.